src/app/modules/main/chat_section/chat_content/input_area/controller.nim

```diff
@@ -22,6 +29,8 @@ type
     gifService: gif_service.Service
     messageService: message_service.Service
     linkPreviewCache: LinkPreviewCache
+    linkPreviewPersistentSetting: LinkPreviewSetting
+    linkPreviewTemporarySetting: LinkPreviewSetting
```

Comment:
> but it really keeps the state of the current message,

Yes, you're right!
I was thinking it's suggestive enough since we have `linkPreviewPersistentSetting` and `linkPreviewTemporarySetting`

`linkPreviewCurrentMessageSetting` could be better actually since this controller now has some logic that's determining the lifetime of this settings as being equal to the current message.